.github/workflows/unit_test_4gpu.yaml

```diff
@@ -31,5 +31,6 @@ jobs:
         pip config --user set global.progress_bar off
 
         python -m pip install --force-reinstall --pre torch --index-url https://download.pytorch.org/whl/nightly/cu121
+        python -m pip install --pre torchdata --index-url https://download.pytorch.org/whl/nightly/
```

Comment:
I think we need this in `.github/workflows/integration_test_periodic.yaml` as well.

Let's create an issue, tracking that we need to put torchdata in `requirements.txt` and `pyproject.toml` after the needed changes ship in an official release.

Reply:
@tianyu-l Should pyproject.toml datasets dependency also enforce >= 2.19.0 version requirement?

Reply:
ah yes I think so

Reply:
Created https://github.com/pytorch/torchtitan/issues/351 to remove torchdata nightly pip install